### PR TITLE
add bridge pallet

### PR DESCRIPTION
**Pull request by lzw65**

### CODE REVIEW

File: pallets/types.json [hunk 1]
```diff
@@ -23,5 +23,70 @@
...
+      "limits": "Limits",
+      "status": "Status"
+    },
+    "BridgeMessage": {
+      "message_id": "H256",
+      "account": "AccountId",
+      "status": "Status",
+      "action": "Status"
+    },
+    "ValidatorsMessage": {
```
Comment (kzjin) [inline pallets/types.json:78]: Should be "ValidatorMessage"

**Review by kzjin — COMMENTED** [review]
Just found one place need to be changed in pallets/types.json.
Change "ValidatorsMessage" to "ValidatorMessage"

File: pallets/types.json (hunk 1 above)
Comment (lzw65) [reply pallets/types.json:78]: I will change it .

File: pallets/bridge/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+        );
+
+        Ok(())
+    }
+    fn check_amount(amount: BalanceOf<T>) -> Result<()> {
+        let max = <CurrentLimits<T>>::get().max_tx_value;
+        let min = <CurrentLimits<T>>::get().min_tx_value;
+
+        ensure!(
+            amount > min,
```
[the remark this answers is not in the document]
Comment (lzw65) [reply pallets/bridge/src/lib.rs:713]: Good idea! I will change it.

File: pallets/bridge/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+              Ok(())
+          }
+
+          // each validator calls it to pause the bridge
+          #[weight = 10_000]
+          pub fn pause_bridge(origin) -> DispatchResult {
+              let validator = ensure_signed(origin)?;
+              Self::check_validator(validator.clone())?;
+
+              ensure!(Self::bridge_is_operational(), "Bridge is not operational already");
```
[the remark this answers is not in the document]
Comment (lzw65) [reply pallets/bridge/src/lib.rs:253]: ensure!(Self::bridge_is_operational(), ... is OK

**Review by fyang1024 — APPROVED** [review]

**Pull request merged**